platform/darwin/src/NSNumber+MGLStyleAttributeAdditions_Private.hpp

```diff
@@ -0,0 +1,15 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLStyleAttributeValue.h"
+
+#include <mbgl/style/property_value.hpp>
+
+@interface NSNumber (MGLStyleAttributeAdditions_Private) <MGLStyleAttributeValue>
```

Comment:
This inheritance strategy looks sound. It’s consistent with our strategy with MGLOfflineRegion and MGLTilePyramidOfflineRegion: MGLTilePyramidOfflineRegion publicly conforms to MGLOfflineRegion and privately conforms to MGLOfflineRegion_Private (which also conforms to MGLOfflineRegion). MGLOfflineStorage imports MGLOfflineRegion_Private.h, so it has access to, for instance, `-initWithOfflineRegionDefinition:` (although it still needs to call `-conformsToProtocol:` first).